### PR TITLE
fix(quote): make quote-link-with-icon available in quote cdn bundle

### PR DESCRIPTION
**Pull request by m4olivei**
### Related Ticket(s)

Relates to https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/10636

### Description

The `<dds-quote-link-with-icon>` component was not included with the main `<dds-quote>` component, leading to a broken experience were a user to attempt to use it with the main quote CDN bundle like the Story example does, for example:

https://codepen.io/m4olivei/pen/jOXdvjK

It looks like there is just a missing import in the `quote/index.ts` file. We add that here and update docs and Sandbox example to use the `<dds-quote-link-with-icon>` component.

### Changelog

**Changed**

- Import `<dds-quote-link-with-icon>` from the main `<dds-quote>` index.ts

### CODE REVIEW

File: packages/web-components/src/components/quote/index.ts [hunk 1]
```diff
@@ -12,3 +12,4 @@ import './quote-source-bottom-copy';
 import './quote-source-copy';
 import './quote-source-heading';
 import '../link-with-icon/link-with-icon';
```
Comment (andy-blum) [inline packages/web-components/src/components/quote/index.ts:14]: Is this element import still needed here since `quote-link-with-icon` is extending it?

Comment (m4olivei) [reply packages/web-components/src/components/quote/index.ts:14]: Ahh yeah quite right. For a second I thought we still needed it for backward compatibility, but we still get it by way ot `quote-link-with-icon`. Will fix.

**Review by jkaeser — APPROVED** [review]
Looks like a good change to me. Not seeing any visual regressions when compared with the latest release.

An argument could be made that the `<dds-quote-link-with-icon>` component isn't strictly necessary... looks like it exists only to have a `display: inline-block;` CSS rule set on it. But that's maybe a conversation for v2 😏

**Review by emyarod — APPROVED** [review]
looks good to me!

**Pull request merged**